builtin/v13/miner/miner_types.go

```diff
@@ -394,3 +394,105 @@ type GetPeerIDReturn struct {
 type GetMultiAddrsReturn struct {
 	MultiAddrs []byte
 }
+
+// ProveCommitSectors3Params represents the parameters for proving committed sectors.
+type ProveCommitSectors3Params struct {
+	SectorActivations          []SectorActivationManifest
+	SectorProofs               [][]byte
+	AggregateProof             []byte
+	AggregateProofType         abi.RegisteredAggregationProof
+	RequireActivationSuccess   bool
+	RequireNotificationSuccess bool
+}
+
+// SectorActivationManifest contains data to activate a commitment to one sector and its data.
+// All pieces of data must be specified, whether or not not claiming a FIL+ activation or being
+// notified to a data consumer.
+// An implicit zero piece fills any remaining sector capacity.
+type SectorActivationManifest struct {
+	SectorNumber abi.SectorNumber
+	Pieces       []PieceActivationManifest
+}
+
+// PieceActivationManifest represents the manifest for activating a piece.
+type PieceActivationManifest struct {
+	CID                   cid.Cid
+	Size                  abi.PaddedPieceSize
+	VerifiedAllocationKey *VerifiedAllocationKey
+	Notify                []DataActivationNotification
+}
+
+// VerifiedAllocationKey represents the key for a verified allocation.
+type VerifiedAllocationKey struct {
+	Client abi.ActorID
+	ID     verifreg.AllocationId
+}
+
+// DataActivationNotification represents a notification for data activation.
+type DataActivationNotification struct {
+	Address addr.Address
+	Payload []byte
+}
+
+// ProveCommitSectors3Return represents the return value for the ProveCommit2 function.
+type ProveCommitSectors3Return = BatchReturn
+
+type BatchReturn struct {
+	SuccessCount uint64
+	FailCodes    []FailCode
+}
+
+type FailCode struct {
+	// Idx represents the index of the operation that failed within the batch.
+	Idx  uint64
+	Code xc.ExitCode // todo correct?
+}
+
+// ProveReplicaUpdates3Params represents the parameters for proving replica updates.
+type ProveReplicaUpdates3Params struct {
+	SectorUpdates              []SectorUpdateManifest
+	SectorProofs               [][]byte
+	AggregateProof             []byte
+	UpdateProofsType           abi.RegisteredUpdateProof
+	AggregateProofType         abi.RegisteredAggregationProof
```

Comment:
```suggestion
	AggregateProofType         *abi.RegisteredAggregationProof
```

probably https://github.com/filecoin-project/builtin-actors/pull/1511